unittests/Dialect/XTenNN/IR/TestXTenNNTypes.cpp

```diff
@@ -0,0 +1,77 @@
+//===-- TestXTenNNTypes.cpp - Test XTenNN Type Definitions*---------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2023 Advanced Micro Devices, Inc.
+//
+//===----------------------------------------------------------------------===//
+
+#include "xten/Dialect/XTenNN/IR/XTenNNTypes.h"
+
+#include <mlir/IR/BuiltinAttributes.h>
+#include <mlir/IR/BuiltinTypes.h>
+
+#include <gtest/gtest.h>
+
+namespace {
+
+struct TestValue {
+  float value;
+  bool expectedResult;
+};
+
+std::ostream &operator<<(std::ostream &os, const TestValue &value) {
+  os << "{ value = " << value.value << ", expected = " << value.expectedResult
+     << " }";
+  return os;
+}
+
+class PowerOfTwoTest : public testing::TestWithParam<TestValue> {};
+
+auto getPowerOfTwoTestCases() {
+  mlir::MLIRContext context;
+  return testing::Values<TestValue>(
+      TestValue({1.0, true}), TestValue({0.125, true}),
```

Comment:
Oh, you are not actually checking that the scale is a power of two, but a reciprocal of a power of two.
Can we rename the functions and comments to say that?

Reply:
Alternatively, let the ops take the `inv_scale` as attribute (the inverse of the scale), which can then be an integer type and actually a power of two.

Reply:
No we check both the variations. I have a test case for `16.0`.